.github/workflows/build.yml

```diff
@@ -0,0 +1,145 @@
+name: build
+
+on:
+  push:
+    branches: [ master, '[1-9][0-9].[0-9]' ]
+  pull_request:
+    branches: [ master, '[1-9][0-9].[0-9]' ]
+
+jobs:
+  build:
+
+    name: "PHP ${{ matrix.php-versions }} Test ${{ matrix.static-analysis != 'no' && matrix.static-analysis || '' }} (deps: ${{ matrix.dependencies }})"
+
+    continue-on-error: ${{ matrix.experimental }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.3', '7.4']
+        dependencies: ['highest']
+        composer-arguments: [''] # to run --ignore-platform-reqs in experimental builds
+        static-analysis: ['no']
+        experimental: [false]
+        include:
+          - php-versions: '7.3'
+            static-analysis: 'psalm'
+            experimental: false
+            dependencies: 'highest'
+
+          # Experimental build for PHP 8
+          #- php-versions: '8.0'
+          #  composer-arguments: '--ignore-platform-reqs'
+          #  static-analysis: 'no'
+          #  experimental: true
+          #  dependencies: 'highest'
+
+          # Build for minimum dependencies. Fails right now, hence deactivated.
+          #- php-versions: '7.3'
+          #  static-analysis: 'no'
+          #  experimental: false
+          #  dependencies: 'lowest'
+
+    runs-on: ubuntu-latest
+    services:
+      mariadb:
```

Comment:
@kdambekalns should we add back a pgsql run? I would add the service here (mapped to a different port), then overwrite `Settings.yaml` after the mysql tests finished and re-run all tests (unit/functional/behat). That should safe us from duplicating all steps for a whole job and avoid spinning up new instances+recompile Flow.

Reply:
It would still recompile, since the configuration changes, but still sounds awesome.